code/obj/item/go_board.dm

```diff
@@ -0,0 +1,196 @@
+					for(var/mob/living/carbon/human/u in src.openwindows)
+						if(u == user)
+							continue
+						if(u.client && !(u in range(u.client.view,src)))
+							u << browse(null, "window=go")
+							break
+						u << output("[href_list["position"]]","go.browser:removepiece")
+					return
+				else if(usr.client && !(usr in range(usr.client.view,src)))
+					usr << browse(null, "window=go")
+					return
+
+	MouseDrop(mob/user as mob)
+		if((istype(user,/mob/living/carbon/human))&&(!user.stat)&&!(src in user.contents))
+			user.put_in_hand_or_drop(src)
+
+	disposing()
+		for(var/mob/living/carbon/human/user in src.openwindows)
+			user << browse(null, "window=go")
+		..()
+
+/obj/item/gobowlb
+	name = "bowl of black stones"
+	desc = "a bowl of little black stones."
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "go-bowl-b"
+
+	attack_hand(mob/user as mob)
+		user.put_in_hand_or_drop(new /obj/item/gostoneb)
+
+	attackby(obj/item/weapon as obj,mob/user as mob)
+		if((istype(weapon, /obj/item/gostoneb)) || (istype(weapon, /obj/item/gostonew)))
+			user.u_equip(weapon)
+			qdel(weapon)
+
+	MouseDrop(mob/user as mob)
+		if((istype(user,/mob/living/carbon/human))&&(!user.stat)&&!(src in user.contents))
+			user.put_in_hand_or_drop(src)
+
+	MouseDrop_T(atom/movable/O as mob|obj, mob/user as mob)
+		var/total
+		var/color
+		if(istype(O,/obj/item/gostoneb))
+			color = "black"
+			for (var/obj/item/gostoneb/s in range(1, user))
+				qdel(s)
+				total++
+				sleep(2)
+		else if(istype(O,/obj/item/gostonew))
+			color = "white"
+			for (var/obj/item/gostonew/s in range(1, user))
+				qdel(s)
+				total++
+				sleep(2)
+		src.visible_message("<span><b>[usr]</b> adds [total] [color] stones to the bowl!</span>")
+
+/obj/item/gobowlw
+	name = "bowl of white stones"
+	desc = "a bowl of little white stones."
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "go-bowl-w"
+
+	attack_hand(mob/user as mob)
+		user.put_in_hand_or_drop(new /obj/item/gostonew)
+
+	attackby(obj/item/weapon as obj,mob/user as mob)
+		if((istype(weapon, /obj/item/gostoneb)) || (istype(weapon, /obj/item/gostonew)))
+			user.u_equip(weapon)
+			qdel(weapon)
+
+	MouseDrop(mob/user as mob)
+		if((istype(user,/mob/living/carbon/human))&&(!user.stat)&&!(src in user.contents))
+			user.put_in_hand_or_drop(src)
+
+	MouseDrop_T(atom/movable/O as mob|obj, mob/user as mob)
+		var/total
+		var/color
+		if(istype(O,/obj/item/gostoneb))
+			color = "black"
+			for (var/obj/item/gostoneb/s in range(1, user))
+				qdel(s)
+				total++
+				sleep(2)
+		else if(istype(O,/obj/item/gostonew))
+			color = "white"
+			for (var/obj/item/gostonew/s in range(1, user))
+				qdel(s)
+				total++
+				sleep(2)
+		src.visible_message("<span><b>[usr]</b> adds [total] [color] stones to the bowl!</span>")
+
+/obj/item/gostoneb
+	name = "black stone"
+	desc = "a little black stone"
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "go-stone-b"
+	w_class = 1
+
+/obj/item/gostonew
```

Comment:
I think these two could also have a common parent `/obj/item/gostone` just for cleanliness.

Reply:
They were originally, but it was more lines of code overall and more of a hassle to work around (seeing as there's only two.) I just felt this was cleaner overall.

Reply:
Also done in next commit